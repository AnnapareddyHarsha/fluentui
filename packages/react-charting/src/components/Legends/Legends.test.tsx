jest.mock('react-dom');
import * as React from 'react';
import { resetIds } from '../../Utilities';
import * as renderer from 'react-test-renderer';
import { Legends } from './index';
import { ILegendState, LegendsBase } from './Legends.base';
import { mount, ReactWrapper } from 'enzyme';
import { DefaultPalette } from '@fluentui/react/lib/Styling';
import { ILegendsProps } from './Legends.types';

// Wrapper of the Legends to be tested.
let wrapper: ReactWrapper<ILegendsProps, ILegendState, LegendsBase> | undefined;

function sharedBeforeEach() {
  resetIds();
}

function sharedAfterEach() {
  if (wrapper) {
    wrapper.unmount();
    wrapper = undefined;
  }

  // Do this after unmounting the wrapper to make sure if any timers cleaned up on unmount are
  // cleaned up in fake timers world
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  if ((global.setTimeout as any).mock) {
    jest.useRealTimers();
  }
}

const legends = [
  {
    title: 'Legend 1',
    color: DefaultPalette.red,
  },
  {
    title: 'Legend 2',
    color: DefaultPalette.black,
  },
];

const styles = {
  rect: {
    borderRadius: '3px',
  },
};

const overflowProps = {
  styles: {
    item: { border: `1px dotted ${DefaultPalette.green}` },
    root: {},
    overflowButton: { backgroundColor: DefaultPalette.neutralLight },
  },
};

const focusZonePropsInHoverCard = {
  'aria-label': 'Legend 1 selected',
};

describe('Legends snapShot testing', () => {
  it('renders Legends correctly', () => {
    const component = renderer.create(<Legends legends={legends} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders allowFocusOnLegends correctly', () => {
    const component = renderer.create(<Legends legends={legends} allowFocusOnLegends={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders canSelectMultipleLegends correctly', () => {
    const component = renderer.create(<Legends legends={legends} canSelectMultipleLegends={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders styles correctly', () => {
    const component = renderer.create(<Legends legends={legends} {...styles} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('AreaChart - basic props', () => {
  beforeEach(sharedBeforeEach);
  afterEach(sharedAfterEach);

  it('Should not mount legends when empty', () => {
    wrapper = mount(<Legends legends={[]} />);
    const legend = wrapper.getDOMNode().querySelectorAll('[class^="legendContainer"]');
    expect(legend!.length).toBe(0);
  });

  it('Should mount legends when not empty', () => {
    wrapper = mount(<Legends legends={legends} />);
    const legend = wrapper.getDOMNode().querySelectorAll('[class^="legendContainer"]');
    expect(legend).toBeDefined();
  });

  it('Should render overflowProps correctly', () => {
    wrapper = mount(<Legends legends={legends} {...overflowProps} />);
    const overflowProp = wrapper.getDOMNode().querySelectorAll('[class^="ms-Layer"]');
    expect(overflowProp).toBeDefined();
  });

  it('Should render data-is-focusable correctly', () => {
    wrapper = mount(<Legends legends={legends} data-is-focusable={true} />);
    const dataIsFocusable = wrapper.getDOMNode().querySelector('[class^="legend"]')?.getAttribute('data-is-focusable');
    expect(dataIsFocusable).toBeTruthy();
  });

  it('Should render focusZonePropsInHoverCard correctly', () => {
    wrapper = mount(<Legends legends={legends} {...focusZonePropsInHoverCard} />);
    const overflowProp = wrapper.getDOMNode().querySelector('[class^="legend"]')?.getAttribute('aria-label');
    expect(overflowProp).toBe('Legend 1 selected');
  });
});

describe('Render calling with respective to props', () => {
  it('No prop changes', () => {
    const renderMock = jest.spyOn(LegendsBase.prototype, 'render');
    const props = {
      legends,
    };
    const component = mount(<Legends {...props} />);
    component.setProps({ ...props });
    expect(renderMock).toHaveBeenCalledTimes(2);
    renderMock.mockRestore();
  });
  it('prop changes', () => {
    const renderMock = jest.spyOn(LegendsBase.prototype, 'render');
    const props = {
      legends,
      allowFocusOnLegends: true,
    };
    const component = mount(<Legends {...props} />);
    component.setProps({ ...props, allowFocusOnLegends: true });
    expect(renderMock).toHaveBeenCalledTimes(2);
    renderMock.mockRestore();
  });
});
