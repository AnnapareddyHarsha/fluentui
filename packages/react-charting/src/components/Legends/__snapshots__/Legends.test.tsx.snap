// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legends snapShot testing renders Legends correctly 1`] = `
<div
  className=

      {
        align-items: center;
        margin-bottom: 0;
        margin-left: -8px;
        margin-right: 0;
        margin-top: -8px;
        white-space: nowrap;
        width: 100%;
      }
>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        data-automation-id="visibleContent"
        style={
          Object {
            "position": "fixed",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-label="Legends"
          className=
              ms-FocusZone
              &:focus {
                outline: none;
              }
          data-focuszone-id="FocusZone0"
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDownCapture={[Function]}
          role="listbox"
        >
          <div
            className=
                ms-OverflowSet
                {
                  display: flex;
                  flex-wrap: wrap;
                  justify-content: unset;
                  position: relative;
                }
            role="group"
          >
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 1 selected"
                aria-posinset={1}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #e81123;
                        border-color: #e81123;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #e81123, #e81123);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 1
                </div>
              </button>
            </div>
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 2 selected"
                aria-posinset={2}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #000000;
                        border-color: #000000;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #000000, #000000);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Legends snapShot testing renders allowFocusOnLegends correctly 1`] = `
<div
  className=

      {
        align-items: center;
        margin-bottom: 0;
        margin-left: -8px;
        margin-right: 0;
        margin-top: -8px;
        white-space: nowrap;
        width: 100%;
      }
>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        data-automation-id="visibleContent"
        style={
          Object {
            "position": "fixed",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-label="Legends"
          className=
              ms-FocusZone
              &:focus {
                outline: none;
              }
          data-focuszone-id="FocusZone1"
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDownCapture={[Function]}
          role="listbox"
        >
          <div
            className=
                ms-OverflowSet
                {
                  display: flex;
                  flex-wrap: wrap;
                  justify-content: unset;
                  position: relative;
                }
            role="group"
          >
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 1 selected"
                aria-posinset={1}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #e81123;
                        border-color: #e81123;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #e81123, #e81123);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 1
                </div>
              </button>
            </div>
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 2 selected"
                aria-posinset={2}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #000000;
                        border-color: #000000;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #000000, #000000);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Legends snapShot testing renders canSelectMultipleLegends correctly 1`] = `
<div
  className=

      {
        align-items: center;
        margin-bottom: 0;
        margin-left: -8px;
        margin-right: 0;
        margin-top: -8px;
        white-space: nowrap;
        width: 100%;
      }
>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        data-automation-id="visibleContent"
        style={
          Object {
            "position": "fixed",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-label="Legends"
          className=
              ms-FocusZone
              &:focus {
                outline: none;
              }
          data-focuszone-id="FocusZone2"
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDownCapture={[Function]}
          role="listbox"
        >
          <div
            className=
                ms-OverflowSet
                {
                  display: flex;
                  flex-wrap: wrap;
                  justify-content: unset;
                  position: relative;
                }
            role="group"
          >
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 1 selected"
                aria-posinset={1}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #e81123;
                        border-color: #e81123;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #e81123, #e81123);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 1
                </div>
              </button>
            </div>
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 2 selected"
                aria-posinset={2}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #000000;
                        border-color: #000000;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #000000, #000000);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Legends snapShot testing renders styles correctly 1`] = `
<div
  className=

      {
        align-items: center;
        margin-bottom: 0;
        margin-left: -8px;
        margin-right: 0;
        margin-top: -8px;
        white-space: nowrap;
        width: 100%;
      }
>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        data-automation-id="visibleContent"
        style={
          Object {
            "position": "fixed",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-label="Legends"
          className=
              ms-FocusZone
              &:focus {
                outline: none;
              }
          data-focuszone-id="FocusZone3"
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDownCapture={[Function]}
          role="listbox"
        >
          <div
            className=
                ms-OverflowSet
                {
                  display: flex;
                  flex-wrap: wrap;
                  justify-content: unset;
                  position: relative;
                }
            role="group"
          >
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 1 selected"
                aria-posinset={1}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #e81123;
                        border-color: #e81123;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #e81123, #e81123);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 1
                </div>
              </button>
            </div>
            <div
              className=
                  ms-OverflowSet-item
                  {
                    display: inherit;
                    flex-shrink: 0;
                  }
              role="none"
            >
              <button
                aria-label="Legend 2 selected"
                aria-posinset={2}
                aria-selected={false}
                aria-setsize={2}
                className=

                    {
                      align-items: center;
                      background: none;
                      border: none;
                      cursor: pointer;
                      display: flex;
                      outline: transparent;
                      padding-bottom: 8px;
                      padding-left: 8px;
                      padding-right: 8px;
                      padding-top: 8px;
                      position: relative;
                      text-transform: capitalize;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid transparent;
                      bottom: 1px;
                      content: "";
                      left: 1px;
                      outline: 1px solid #605e5c;
                      position: absolute;
                      right: 1px;
                      top: 1px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){.ms-Fabric--isFocusVisible &:focus:after {
                      outline-color: #605e5c;
                    }
                data-is-focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                role="option"
              >
                <div
                  className=

                      {
                        background-color: #000000;
                        border-color: #000000;
                        border: 1px solid;
                        content: ;
                        height: 12px;
                        margin-right: 8px;
                        width: 12px;
                      }
                      @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                        content: linear-gradient(to right, #000000, #000000);
                        opacity: ;
                      }
                />
                <div
                  className=

                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        color: #323130;
                        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                        font-size: 12px;
                        font-weight: 400;
                        line-height: 16px;
                        opacity: ;
                      }
                >
                  Legend 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
